File: BinarySearch/BinarySearch_test.py
import unittest
from BinarySearch import BinarySearchRecursive

class TestBinarySearch(unittest.TestCase):
    
    def test_binary_search_from_file(self):
        binary_search = BinarySearchRecursive()
        
        with open('BinarySearchData.txt', 'r') as file:
            for line in file:
                parts = line.strip().split('|')
                
                arr = list(map(int, parts[0].strip().split(',')))
                
                key = int(parts[1].strip())
                expected_result = int(parts[2].strip())
                
                actual_result = binary_search.binary_search(arr, key, 0, len(arr) - 1)
                
                self.assertEqual(expected_result, actual_result, f"Binary search failed for input: {arr} and key: {key}")

if __name__ == "__main__":
    unittest.main()

File: BinarySearch/BinarySearchTest.java
package BinarySearch;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;
import java.io.*;
import java.util.*;

public class BinarySearchTest {

    @Test
    public void testBinarySearchFromFile() throws IOException {
        BinarySearch_Recursive binarySearch = new BinarySearch_Recursive();
        BufferedReader reader = new BufferedReader(new FileReader("BinarySearchData.txt"));
        String line;

        while ((line = reader.readLine()) != null) {
            String[] parts = line.split("\\|");

            // Parse array
            String[] arrStr = parts[0].trim().split(",");
            int[] arr = new int[arrStr.length];
            for (int i = 0; i < arrStr.length; i++) {
                arr[i] = Integer.parseInt(arrStr[i].trim());
            }

            int key = Integer.parseInt(parts[1].trim());

            int expectedResult = Integer.parseInt(parts[2].trim());

            int actualResult = binarySearch.binarySearch(arr, key, 0, arr.length - 1);

            assertEquals(expectedResult, actualResult, "Binary search failed for input: " + Arrays.toString(arr) + " and key: " + key);
        }

        reader.close();
    }
}
